File: bench/type4/adversarial/cases/null_option_presence/default.rs
pub fn rust_unwrap_or(
    value: Option<i32>,
    fallback: i32,
    other: Option<i32>,
    other_default: i32,
) -> i32 {
    value.unwrap_or(fallback)
}

pub fn rust_match_default(
    value: Option<i32>,
    fallback: i32,
    other: Option<i32>,
    other_default: i32,
) -> i32 {
    match value {
        Some(inner) => inner,
        None => fallback,
    }
}

pub fn rust_wrong_value(
    value: Option<i32>,
    fallback: i32,
    other: Option<i32>,
    other_default: i32,
) -> i32 {
    other.unwrap_or(fallback)
}

pub fn rust_wrong_fallback(
    value: Option<i32>,
    fallback: i32,
    other: Option<i32>,
    other_default: i32,
) -> i32 {
    value.unwrap_or(other_default)
}

File: bench/type4/adversarial/cases/null_option_presence/default.swift
func swiftCoalesce(_ value: Int?, _ fallback: Int, _ other: Int?, _ otherDefault: Int) -> Int {
    return value ?? fallback
}

func swiftWrongFallback(_ value: Int?, _ fallback: Int, _ other: Int?, _ otherDefault: Int) -> Int {
    return value ?? otherDefault
}

func swiftWrongValue(_ value: Int?, _ fallback: Int, _ other: Int?, _ otherDefault: Int) -> Int {
    return other ?? fallback
}

func expensive() -> Int {
    return 1
}

func swiftEffectfulFallback(_ value: Int?, _ fallback: Int, _ other: Int?) -> Int {
    return value ?? expensive()
}
